File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/brandService.js
// 品牌服务层
app.service("brandService",function($http){
	
	// 查询所有品牌
	this.findAll=function(){
		return $http.get("../brand/findAll.do");
	}
	
	// 添加品牌
	this.add=function(entity){
		return $http.post("../brand/add.do",entity);
	}
	
	// 修改品牌
	this.update=function(entity){
		return $http.post("../brand/update.do",entity);
	}
	
	// 通过id查找用户
	this.findOne=function(id){
		return $http.get("../brand/findOne.do?id="+id);
	}
	
	// 批量删除
	this.dele=function(selectIds){
		return $http.get("../brand/delete.do?ids="+selectIds);
	}
	
	// 条件搜索
	this.search=function(page,size,searchEntity){
		return $http.post("../brand/search.do?page="+page+"&size="+size,searchEntity);
	}
	
	// 下拉列表数据
	this.selectOptionList=function(){
		return $http.get("../brand/selectOptionList.do");
	}
	
})

File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.pinyougou.pojo.TbBrand;

import entity.PageResult;
import entity.Result;

/**
 * 品牌接口
 * @author hwc
 *
 */
public interface BrandService {
	
	
	public List<TbBrand> findAll();
	
	/**
	 * 分页查询所有品牌
	 * @param pageNum 当前页码
	 * @param pageSize 每页显示的条数
	 * @return
	 */
	public PageResult<TbBrand> findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加品牌
	 * @param brand
	 */
	public void add(TbBrand brand);
	
	/**
	 * 通过id查找品牌
	 * @param id
	 * @return
	 */
	public TbBrand findOne(Long id);
	
	/**
	 * 修改品牌信息
	 * @param brand
	 */
	public void update(TbBrand brand);
	
	/**
	 * 删除品牌
	 * @param ids
	 */
	public void delete(Long[] ids);
	
	
	/**
	 * 条件分页查询
	 * @param brand
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	public PageResult<TbBrand> findPage(TbBrand brand,int pageNum, int pageSize);
	
	/**
	 * 品牌下拉框数据
	 * @return
	 */
	List<Map> selectOptionList();
}
